LibraBFT/Impl/Consensus/Types/EpochIndep.agda

```diff
@@ -257,12 +257,22 @@ module LibraBFT.Impl.Consensus.Types.EpochIndep where
              (siHighestQuorumCert ∷ siHighestCommitCert ∷ [])
   postulate instance enc-SyncInfo : Encoder SyncInfo
 
+  {- TODO-HC how do if-then-else and equality check?
```

Comment:
Please change `HC-TODO` to `TODO-1` (see the [`TODO` guide](https://github.com/oracle/bft-consensus-agda/blob/main/TODO.md)).  Also, here are some suggestions to help you make progress on the `TODO`, if you like.

Unlike Haskell, we cannot simply say `deriving Eq`; we need to define equality for `QuorumCert`.  That is straightforward and slightly tedious.  I've postulated it below (`_≟QC_`), and then shown how we can use it to define what you want.
```
  record SyncInfo : Set where
    field
      ₋siHighestQuorumCert  : QuorumCert
      ₋siHighestCommitCert  : Maybe QuorumCert
      -- ₋siHighestTimeoutCert : Mabe TimeoutCert -- Not used yet.
  open SyncInfo public
  unquoteDecl siHighestQuorumCert   = mkLens (quote SyncInfo)
             (siHighestQuorumCert ∷ [])
  postulate instance enc-SyncInfo : Encoder SyncInfo

  postulate -- TODO-1: define this, it should be easy, but requires defining Dec instances for Vote
            -- Data, BlockInfo, and KVMap.
    _≟QC_ : (q1 q2 : QuorumCert) → Dec (q1 ≡ q2)

  _QCBoolEq_ : QuorumCert → QuorumCert → Bool
  _QCBoolEq_ q1 q2 = does (q1 ≟QC q2)

  SyncInfo∙new : QuorumCert → QuorumCert → SyncInfo
  SyncInfo∙new highestQuorumCert highestCommitCert =
    record { ₋siHighestQuorumCert = highestQuorumCert
           ; ₋siHighestCommitCert = if highestQuorumCert QCBoolEq highestCommitCert
                                    then nothing else (just highestCommitCert) }

  siHighestCommitCert : Lens SyncInfo QuorumCert
  siHighestCommitCert = mkLens' (λ x → maybe id (x ^∙ siHighestQuorumCert) (₋siHighestCommitCert x))
                                (λ x si → record x { ₋siHighestCommitCert = just si })

```
Note that I removed the "automatic" lens for `siHighestCommitCert` so I could replace it with the manual one above.  Also I made `₋siHighestCommitCert  : Maybe QuorumCert`, which I think is what we want, right?

Reply:
Actually, I found a more convenient way using the (postulated, for now) fact that we have an injective encoder for `QuorumCert`; see #27.